### DISCUSSION TITLE
Duplicated execution in Allure report

### DISCUSSION THREAD

**Discussion by ciomax** [allure-framework/allure2#1891] [Questions & Support] [answered]
Hi,

I am using Allure with TestNg, and the issue is that if I execute the same test twice but with different parameters in testng, I will se different executions in the report.
How can i fix this?

![image](https://user-images.githubusercontent.com/58025896/212342134-7e2f3e6a-8b37-4462-bb3c-af7327ac7e67.png)

**Comment by cheshi-mantu** [comment]
hi @ciomax this is correct behaviour. Test results you get for different values of your parameters are different, i.e. this cannot be considered as a retry but as a new test result.

**Reply by lzhinin** [reply]
Hi there,

I have a question related to the same topic. In my scenario, I've noticed duplicate results for a particular test case that has no parameters. The distinguishing factor is that one of the test cases fails, resulting in fewer steps in the Test Body. Could you clarify if the Test Body also influences how test case results are grouped? I am running Allure with pytest. 
![image](https://github.com/allure-framework/allure2/assets/91629361/7e91c93e-2786-4dc2-9460-09a3b4939718)

**Reply by baev** [reply]
Please read the [History & Retries](https://allurereport.org/docs/history-and-retries/#retries) article first; we tried our best to describe how it works.

**Reply by lzhinin** [reply]
Thanks for the quick response! I did look at this first. However, that documentations says "Normally, when you run a test with a set of parameters different from the previous times, Allure Report treats it as a new test". In my case I am not using any parameters, unless the Allure steps are considered "parameters". Can you please clarify?

**Reply by baev** [reply]
The behaviour is most likely related to the way retries are implemented. For example, some test frameworks don't trigger proper test/start events for retried tests. In that scenario, Allure tries its best to make results as consistent as possible. 

Either way, please don't hesitate to open a separate discussion. We'll need to know the versions (python, pytest, allure-pytest, allure-commandline) and a bit of information on how you get your duplicate results. 

The best way — is to share a project repo to reproduce the problem (as GitHub repo).

**Comment by baev** [comment, marked as answer]
If you want results to be merged, mark the parameter as excluded. See https://allurereport.org/docs/testng-reference/#parametrized-tests and https://allurereport.org/docs/history-and-retries/
